docs/advanced/hyperparameter-tuning.md

```diff
@@ -0,0 +1,211 @@
+# Hyperparameter Tuning
+
+CleanRL comes with a simple and practical hyperparameter tuning utility `Tuner`.
+
+## Get started
+
+Create the following file:
+
+```python title="tuner_example.py"
+from cleanrl_utils.tuner import Tuner
+tuner = Tuner(
+    script="cleanrl/ppo.py",
+    metric="charts/episodic_return",
+    metric_last_n_average_window=50,
+    direction="maximize",
+    target_scores={
+        "CartPole-v1": [0, 500],
+    },
+    params_fn=lambda trial: {
+        "learning-rate": trial.suggest_loguniform("learning-rate", 0.0003, 0.003),
+        "num-minibatches": trial.suggest_categorical("num-minibatches", [1, 2, 4]),
+        "update-epochs": trial.suggest_categorical("update-epochs", [1, 2, 4]),
+        "num-steps": trial.suggest_categorical("num-steps", [5, 16, 32, 64, 128]),
+        "vf-coef": trial.suggest_uniform("vf-coef", 0, 5),
+        "max-grad-norm": trial.suggest_uniform("max-grad-norm", 0, 5),
+        "total-timesteps": 10000,
+        "num-envs": 4,
+    },
+)
+tuner.tune(
+    num_trials=100,
+    num_seeds=3,
+)
+```
+
+Then you can run the tuner with 
+
+```bash
+poetry install optuna
+python tuner_example.py
+```
+
+Here is what happened:
+
+1. The `tuner_example.py` launches `num_trials=100` *trials* to find the best single set of hyperparameters for `CartPole-v1` in `script="cleanrl/ppo.py"`.
+1. Each *trial* uses a set of hyperparameters sampled from the `params_fn` to run `num_seeds=3` *experiments* with different random seeds, mitigating the impact of randomness on the results. 
+    * In each *experiment*, `tuner_example.py` averages the last `metric_last_n_average_window=50` reported  `metric="charts/episodic_return"` to a number $x_i$ and calculate a normalized score  $z_i = (x_i - 0) / (500 - 0)$ according to `target_scores`. 
```

Comment:
For calculating the normalized score. maybe change 0, 500 to something like targe_score.min, target_score.max?